docs/source/execute_api.rst

```diff
@@ -0,0 +1,145 @@
+Executing notebooks
+===================
+
+In this section we show how to execute a ``.ipynb`` notebook
+document saving the result in notebook format.
+To export the notebook to other formats see the
+:ref:`next section <nbconvert_library>`.
+
+Executing notebooks programmatically is useful, for example, as a test layer
+in python libraries that include example notebooks, or as a way to
+automate the data analysis in projects involving more than one notebook.
+
+The same functionality of executing notebooks is exposed through a
+:ref:`command line interface <commandline_usage>` or a python API interface.
+In this section we will (mostly) illustrate how to use the python API interface.
+
+A Quick example
+---------------
+
+Let's start with a complete quick example, leaving detailed explanations
+to the following sections.
+
+First we import nbconvert and the ``ExecutePreprocessor`` class:
+
+.. code-block:: python
+
+    import nbformat
+    from nbconvert.preprocessors import ExecutePreprocessor
+
+Assuming that ``notebook_filename`` is the path of a notebook,
+we can load it with:
+
+.. code-block:: python
+
+    nb = nbformat.read(open(notebook_filename), as_version=4)
+
+Next, we configure the notebook execution mode:
+
+.. code-block:: python
+
+    ep = ExecutePreprocessor(timeout=3600, kernel_name='python3')
+
+We specified two (optional) arguments ``timeout`` and ``kernel_name``, which
+define respectively the execution timeout and the execution kernel.
+
+    The **kernel_name** keyword `requires <https://github.com/jupyter/nbconvert/pull/177>`__
+    nbconvert 4.2 (unreleased, use master branch from github).
+
+To actually run the notebook we call the method ``preprocess``:
+
+.. code-block:: python
+
+    ep.preprocess(nb, {'metadata': {'path': 'notebooks/'}})
+
+Hopefully, we will not get any errors during the notebook execution
+(see the last section for error handling). Note that ``path`` specifies
+in which folder to execute the notebook.
+Finally, to save the resulting notebook:
+
+.. code-block:: python
+
+    nbformat.write(nb, open('executed_notebook.ipynb', mode='wt'))
```

Comment:
Let's use a context manager:

```python
with open('executed_notebook.ipynb', 'wt') as f:
    nbformat.write(nb, f)
```

I like to follow best practices in code examples for docs much more rigorously than I do in actual code files, because code examples will end up being copied and pasted everywhere.